src/Symfony/Bundle/SecurityBundle/Debug/WrappedLazyListener.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Debug;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\Security\Core\Exception\LazyResponseException;
+use Symfony\Component\Security\Http\Firewall\AbstractListener;
+use Symfony\Component\Security\Http\Firewall\ListenerInterface;
+use Symfony\Component\VarDumper\Caster\ClassStub;
+
+/**
+ * Wraps a lazy security listener.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @internal
+ */
+final class WrappedLazyListener extends AbstractListener implements ListenerInterface
+{
+    use TraceableListenerTrait;
+
+    public function __construct(AbstractListener $listener)
+    {
+        $this->listener = $listener;
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $this->listener->supports($request);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function authenticate(RequestEvent $event)
+    {
+        $startTime = microtime(true);
+
+        try {
+            $ret = $this->listener->authenticate($event);
+        } catch (LazyResponseException $e) {
+            $this->response = $e->getResponse();
+
+            throw $e;
+        } finally {
+            $this->time = microtime(true) - $startTime;
+        }
+
+        $this->response = $event->getResponse();
+
+        return $ret;
+    }
+
+    public function getInfo(): array
+    {
+        if (null !== $this->stub) {
+            // no-op
+        } elseif (self::$hasVarDumper ?? self::$hasVarDumper = class_exists(ClassStub::class)) {
```

Comment:
we can even remove this check as var-dumper is a mandatory transitive deps now (same in WrappedListener)

Reply:
done